### ISSUE TITLE
dbengine files not deleting

### ISSUE THREAD

**Issue by davent** [netdata/netdata#7874]
##### Bug report summary
Old dbengine files are not being deleted. The /var/cache/netdata/dbengine directory is filling out of control due to old files not being cleaned up/deleted. Once this occurs new files are created but all existing files remain slowly filling the disk.

The old files will be cleaned up on restart of netdata.

A successful creation and deletion of the dbengine files prior to the error occuring can be seen in the logs:
```bash
2019-12-31 01:22:56: netdata INFO  : MAIN : Creating new data and journal files in path /var/cache/netdata/dbengine
2019-12-31 01:22:56: netdata INFO  : MAIN : Created data file "/var/cache/netdata/dbengine/datafile-1-0000000090.ndf".
2019-12-31 01:22:56: netdata INFO  : MAIN : Created journal file "/var/cache/netdata/dbengine/journalfile-1-0000000090.njf".
```
and later successfully deleted: 
```bash
2020-01-01 19:45:52: netdata INFO  : MAIN : Deleting data file "/var/cache/netdata/dbengine/datafile-1-0000000090.ndf".
2020-01-01 19:45:52: netdata INFO  : MAIN : Deleting data and journal file pair.
2020-01-01 19:45:52: netdata INFO  : MAIN : Deleted journal file "/var/cache/netdata/dbengine/journalfile-1-0000000090.njf".
2020-01-01 19:45:52: netdata INFO  : MAIN : Deleted data file "/var/cache/netdata/dbengine/datafile-1-0000000090.ndf".
2020-01-01 19:45:52: netdata INFO  : MAIN : Reclaimed 17780736 bytes of disk space.
```
However, the next file pair:
```bash
2019-12-31 05:04:48: netdata INFO  : MAIN : Creating new data and journal files in path /var/cache/netdata/dbengine
2019-12-31 05:04:48: netdata INFO  : MAIN : Created data file "/var/cache/netdata/dbengine/datafile-1-0000000091.ndf".
2019-12-31 05:04:48: netdata INFO  : MAIN : Created journal file "/var/cache/netdata/dbengine/journalfile-1-0000000091.njf".
```
are created but then the only line relating to their deletion is:
```bash
2020-01-01 22:19:28: netdata INFO  : MAIN : Deleting data file "/var/cache/netdata/dbengine/datafile-1-0000000091.ndf".
```
No further mention of this file is seen in the logs.

Also, the next file pair in the sequence is not seen to be created (0000000092) at all and all further files, 0000000093 onwards, are seen to be created but no attempts to delete them are made.

##### OS / Environment
```
/etc/lsb-release:DISTRIB_ID=Ubuntu
/etc/lsb-release:DISTRIB_RELEASE=16.04
/etc/lsb-release:DISTRIB_CODENAME=xenial
/etc/lsb-release:DISTRIB_DESCRIPTION="Ubuntu 16.04.6 LTS"
/etc/os-release:NAME="Ubuntu"
/etc/os-release:VERSION="16.04.6 LTS (Xenial Xerus)"
/etc/os-release:ID=ubuntu
/etc/os-release:ID_LIKE=debian
/etc/os-release:PRETTY_NAME="Ubuntu 16.04.6 LTS"
/etc/os-release:VERSION_ID="16.04"
/etc/os-release:HOME_URL="http://www.ubuntu.com/"
/etc/os-release:SUPPORT_URL="http://help.ubuntu.com/"
/etc/os-release:BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
/etc/os-release:VERSION_CODENAME=xenial
/etc/os-release:UBUNTU_CODENAME=xenial
```

##### Netdata version
 netdata v1.18.1

##### Component Name
dbengine

##### Steps To Reproduce
(unknown at this time)

##### Expected behavior
When the /var/cache/netdata/dbengine directory reaches 256MB in size, old file pairs should be successfully deleted.

**Comment by mfundul** [comment]
@davent there has been a bug with identical behaviour that has been fixed before. You should upgrade to the latest code and see if this fixes the problem for you.

**Comment by davent** [comment]
@mfundul I could not find any bugs like this. Please could you provide the the issue number so that I can test the version containing the fix please?

**Comment by mfundul** [comment]
@davent the commit version you should test is anything newer than #7565
It is kind of complicated to keep track of what exactly happened with the connected issues, but you can examine the description if you like.

**Comment by davent** [comment]
@mfundul I do not see a direct relation between the issues I witness and that merge but maybe still related?

As this fix is unreleased atm it will be harder to test as we are only seeing the issue on a handful of servers out of thousands. Therefore I will need to wait for the next release containing this branch before I can safely upgrade enough machines to see if it has any affect.

**Comment by mfundul** [comment]
It's related based on private logs that I examined, but there is no public visibility unfortunately. You could try the nightly builds to test this, or wait, your choice.

**Comment by amoss** [comment]
@mfundul Please triage this issue. If it can be replicated then we can talk about adding a fix into the next sprint. If it cannot be replicated then please close with a request to reopen when replication instructions are available.

**Comment by mfundul** [comment]
Should have been fixed by anything newer than #7565. Please reopen the issue if the problem persists.
